server/app.js

```diff
@@ -63,4 +189,219 @@ io.on('err', function(err){
 
 http.listen(port,function(){
     log.info('Server listening through port %s', port);
-});
\ No newline at end of file
+});
+
+http.on('err', function(err){
```

Comment:
This and below can probably be separated to a different file.

Reply:
Not sure on this one, did you want the separate file to be somehow included? Might be better to keep it as is for code solidarity